### PR TITLE
Refactor data models to be generic to data storage

### PR DESCRIPTION
**Pull request by ykmr1224**
### Description
-  Refactor data models to be generic to data storage
 - Replaced OpenSearch specific attributes with generic metadata map
 - Replaced some constructor with builder (mainly for readability)

### Issues Resolved
n/a
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: spark/src/test/java/org/opensearch/sql/spark/asyncquery/AsyncQueryExecutorServiceImplTest.java [hunk 1]
```diff
@@ -44,6 +44,9 @@
 @ExtendWith(MockitoExtension.class)
 public class AsyncQueryExecutorServiceImplTest {
 
+  public static final String APPLICATION_ID = "00fd775baqpu4g0p";
+  public static final String EMR_JOB_EXECUTION_ROLE_ARN =
+      "arn:aws:iam::270824043731:role/emr-job-execution-role";
```
Comment (seankao-az) [inline spark/src/test/java/org/opensearch/sql/spark/asyncquery/AsyncQueryExecutorServiceImplTest.java:49]: curious: can these be TestConstants.EMRS_APPLICATION_ID and EMRS_EXECUTION_ROLE?

File: spark/src/test/java/org/opensearch/sql/spark/flint/operation/FlintIndexOpTest.java [hunk 2]
```diff
@@ -33,54 +34,53 @@ public class FlintIndexOpTest {
...
+
     IllegalStateException illegalStateException =
         Assertions.assertThrows(IllegalStateException.class, () -> flintIndexOp.apply(metadata));
+
     Assertions.assertEquals(
         "Moving to transition state:DELETING failed.", illegalStateException.getMessage());
   }
 
+  @NotNull
+  private static FlintIndexStateModel getFlintIndexStateModel(FlintIndexMetadata metadata) {
```
Comment (seankao-az) [inline spark/src/test/java/org/opensearch/sql/spark/flint/operation/FlintIndexOpTest.java:53]: nit: please move to beginning/end of class.. now it's between two test cases

**Review by seankao-az — DISMISSED** [review]

File: spark/src/test/java/org/opensearch/sql/spark/asyncquery/AsyncQueryExecutorServiceImplSpecTest.java [hunk 3]
```diff
@@ -193,7 +194,9 @@ public void reuseSessionWhenCreateAsyncQuery() {
         2,
         search(
             QueryBuilders.boolQuery()
-                .must(QueryBuilders.termQuery("type", STATEMENT_DOC_TYPE))
+                .must(
+                    QueryBuilders.termQuery(
+                        "type", StatementModelXContentSerializer.STATEMENT_DOC_TYPE))
```
[the remark this answers is not in the document]
Comment (ykmr1224) [reply spark/src/test/java/org/opensearch/sql/spark/asyncquery/AsyncQueryExecutorServiceImplSpecTest.java]: Not intentionally, fixed.

File: spark/src/test/java/org/opensearch/sql/spark/asyncquery/AsyncQueryExecutorServiceImplTest.java (hunk 1 above)
Comment (ykmr1224) [reply spark/src/test/java/org/opensearch/sql/spark/asyncquery/AsyncQueryExecutorServiceImplTest.java]: Good point. Fixed.

File: spark/src/test/java/org/opensearch/sql/spark/flint/operation/FlintIndexOpTest.java (hunk 2 above)
Comment (ykmr1224) [reply spark/src/test/java/org/opensearch/sql/spark/flint/operation/FlintIndexOpTest.java]: Fixed.

File: spark/src/test/java/org/opensearch/sql/spark/asyncquery/AsyncQueryExecutorServiceImplTest.java [hunk 4]
```diff
@@ -90,7 +88,7 @@ void testCreateAsyncQuery() {
         .storeJobMetadata(
             AsyncQueryJobMetadata.builder()
                 .queryId(QUERY_ID)
-                .applicationId(APPLICATION_ID)
+                .applicationId(EMRS_APPLICATION_ID)
                 .jobId(EMR_JOB_ID)
                 .build());
```
Comment (seankao-az) [inline spark/src/test/java/org/opensearch/sql/spark/asyncquery/AsyncQueryExecutorServiceImplTest.java:93]: now this can be getAsyncQueryJobMetadata()

**Review by seankao-az — APPROVED** [review]
thanks!

**Review by rupal-bq — APPROVED** [review]

**Pull request merged**